corehq/apps/importer/util.py

```diff
@@ -95,6 +97,25 @@ def from_request(cls, request):
 
         )
 
+
+def open_workbook(filename=None, **kwargs):
+    _soft_assert = soft_assert(notify_admins=True)
+    try:
+        return xlrd.open_workbook(filename=filename, **kwargs)
+    except xlrd.XLRDError as e:
+        message = unicode(e)
+        if message == u'Workbook is encrypted':
+            raise ImporterExcelFileEncrypted(message)
+        else:
+            _soft_assert(False, 'displaying XLRDError directly to user', e)
+            raise ImporterExcelError(message)
+    except IOError as e:
+        raise ImporterFileNotFound(unicode(e))
+    except Exception as e:
+        _soft_assert(False, 'xlrd.open_workbook raised unaccounted for error', e)
+        raise ImporterFileNotFound(unicode(e))
```

Comment:
Soft assert will send you the traceback, so no need to worry about that. I'm leaving the catchall raising `ImporterFileNotFound` because that maintains the previous behavior (the error message that got displayed when this happened was the one I'm now showing for `ImporterFileNotFound`). I pulled out `IOError` because I'm guessing that's what `Exception` was attempting to catch (and is definitely what's raised if you pass in a path to a nonexistent file), but I'm not certain that's the only thing being caught. So, I'm emulating previous behavior and added the soft assert because I'm not happy with it and want to know what the right way to change it is.